ios/FluentUI/DrawerVnext/Drawer.swift

```diff
@@ -0,0 +1,331 @@
+                            radius: tokens.shadowBlur,
+                            x: tokens.shadowDepth[0],
+                            y: tokens.shadowDepth[1])
+                    .offset(x: computedOffset(screenSize: proxy.portraitOrientationAgnosticSize()))
+                    .animation(backgroundLayerAnimation)
+
+                if state.presentationDirection == .left {
+                    InteractiveSpacer().onTapGesture {
+                        state.isExpanded.toggle()
+                    }
+                }
+            }
+            .background(state.isExpanded ? backgroundLayerColor.opacity(backgroundLayerOpacity) : Color.clear)
+            .gesture(dragGesture(snapWidth: proxy.portraitOrientationAgnosticSize().width * percentSnapWidthOfScreen))
+        }
+    }
+
+    private func dragGesture(snapWidth: CGFloat) -> some Gesture {
+        DragGesture()
+            .onChanged { value in
+                let withinDragBounds = state.presentationDirection == .left ? value.translation.width < 0 : value.translation.width > 0
+                if withinDragBounds {
+                    draggedOffsetWidth = value.translation.width
+                }
+            }
+            .onEnded { _ in
+                if let draggedOffsetWidth = draggedOffsetWidth {
+                    if abs(draggedOffsetWidth) < snapWidth {
+                        state.isExpanded = true
+                    } else {
+                        state.isExpanded = false
+                    }
+                    self.draggedOffsetWidth = nil
+                }
+            }
+    }
+}
+
+extension GeometryProxy {
+    func portraitOrientationAgnosticSize() -> CGSize {
+        let isPortraitMode = self.size.width < self.size.height
+        if isPortraitMode {
+            return CGSize(width: self.size.width, height: self.size.height)
+        } else {
+            return CGSize(width: self.size.height, height: self.size.width)
+        }
+    }
+}
+
+public struct InteractiveSpacer: View {
+    public var body: some View {
+        ZStack {
+            Color.black.opacity(0.001)
+        }
+    }
+}
+
+// Mark - Drawer Preview SwiftUI
+
+struct DrawerContent: View {
+    var body: some View {
+        ZStack {
+            Color.red
+            Text("Tap outside to collapse.")
+        }
+    }
+}
+
+struct DrawerPreview: View {
+    var drawer = Drawer(content: DrawerContent())
+    var body: some View {
+        ZStack {
+            NavigationView {
+                EmptyView()
+                    .navigationBarTitle(Text("Drawer Background"))
+                    .navigationBarItems(leading: Button(action: {
+                        drawer.state.isExpanded.toggle()
+                    }, label: {
+                        Image(systemName: "sidebar.left")
+                    })).background(Color.blue)
+            }
+            drawer
```

Comment:
drawer.edgesIgnoringSafeArea(.all)

(so we can display the drawer in the whole screen?)

Reply:
This is already part of `DrawerShimView`.